examples/apps/branching/src/model/groceryList.ts

```diff
@@ -0,0 +1,174 @@
+		public readonly branch: () => Promise<IGroceryList>,
+	) {
+		if (this.disposableParent.disposed) {
+			this.dispose();
+		} else {
+			this.disposableParent.once("dispose", this.dispose);
+			this.map.on("valueChanged", this.onMapValueChanged);
+
+			for (const [id, groceryName] of this.map) {
+				const preExistingGroceryItem = new GroceryItem(id, groceryName, () => {
+					this.map.delete(id);
+				});
+				this._groceryItems.set(id, preExistingGroceryItem);
+			}
+		}
+	}
+
+	public readonly addItem = (name: string) => {
+		this.map.set(uuid(), name);
+	};
+
+	public readonly getItems = (): IGroceryItem[] => {
+		return [...this._groceryItems.values()];
+	};
+
+	private readonly onMapValueChanged = (changed: IValueChanged) => {
+		const changedId = changed.key;
+		const newName = this.map.get(changedId);
+		if (newName === undefined) {
+			this._groceryItems.delete(changedId);
+			this._events.emit("itemDeleted");
+		} else {
+			const newGroceryItem = new GroceryItem(changedId, newName, () => {
+				this.map.delete(changedId);
+			});
+			this._groceryItems.set(changedId, newGroceryItem);
+			this._events.emit("itemAdded");
+		}
+	};
+
+	/**
+	 * Called when the host container closes and disposes itself
+	 */
+	private readonly dispose = (): void => {
+		this._disposed = true;
+		this.map.off("valueChanged", this.onMapValueChanged);
+		this._events.emit("disposed");
+	};
+}
+
+const mapId = "grocery-list";
+
+const mapFactory = new MapFactory();
+const groceryListSharedObjectRegistry = new Map<string, IChannelFactory>([
+	[mapFactory.type, mapFactory],
+]);
+
+export class GroceryListFactory implements IFluidDataStoreFactory {
+	public get type(): string {
+		throw new Error("Do not use the type on the data store factory");
+	}
+
+	public get IFluidDataStoreFactory(): IFluidDataStoreFactory {
+		return this;
+	}
+
+	public async instantiateDataStore(
+		context: IFluidDataStoreContext,
+		existing: boolean,
+	): Promise<IFluidDataStoreChannel> {
+		const runtime: FluidDataStoreRuntime = new FluidDataStoreRuntime(
+			context,
+			groceryListSharedObjectRegistry,
+			existing,
+			async () => instance,
+		);
+
+		let map: ISharedMap;
+		if (existing) {
+			map = (await runtime.getChannel(mapId)) as ISharedMap;
+		} else {
+			map = runtime.createChannel(mapId, mapFactory.type) as ISharedMap;
+			map.set(uuid(), "apple");
+			map.set(uuid(), "banana");
+			map.set(uuid(), "chocolate");
+			map.bindToContext();
+		}
+
+		assert(runtime.entryPoint !== undefined, "EntryPoint was undefined");
+		const handle = runtime.entryPoint;
+
+		// TODO: Use actual branching.  This is currently just creating a detached map and copying the data over.
+		const branchMap = (originalMap: ISharedMap) => {
+			const branchedMap = runtime.createChannel(uuid(), mapFactory.type) as ISharedMap;
+			for (const [key, value] of originalMap) {
+				branchedMap.set(key, value);
+			}
+			return branchedMap;
+		};
+
+		const branch = async () => {
```

Comment:
Fake branching here.  I think it should be fairly straightforward to integrate @anthony-murphy 's prototype here AFAICT, but I haven't actually tried.